custom_components/tahoma/climate_devices/atlantic_pass_apcdhw.py

```diff
@@ -0,0 +1,178 @@
+CORE_DWH_ON_OFF_STATE = "core:DHWOnOffState"
+CORE_ECO_TARGET_DWH_TEMPERATURE_STATE = "core:EcoTargetDHWTemperatureState"
+CORE_STATUS_STATE = "core:StatusState"
+CORE_TARGET_DWH_TEMPERATURE_STATE = "core:TargetDHWTemperatureState"
+
+CUSTOM_PRESET_OFF = "Off"
+CUSTOM_PRESET_PROG = "Prog"
+
+DWH_OFF_STATE = "off"
+DHW_ON_STATE = "on"
+
+IO_PASS_APCDHW_CONFIGURATION_STATE = "io:PassAPCDHWConfigurationState"
+IO_PASS_APCDWH_MODE_STATE = "io:PassAPCDHWModeState"
+IO_PASS_APCDHW_PROFILE_STATE = "io:PassAPCDHWProfileState"
+
+PASS_APCDHW_MODE_COMFORT = "comfort"
+PASS_APCDHW_MODE_ECO = "eco"
+PASS_APCDWH_MODE_INTERNAL_SCHEDULING = "internalScheduling"
+PASS_APCDHW_MODE_STOP = "stop"
+
+
+MAP_HVAC_MODES = {
+    DHW_ON_STATE: HVAC_MODE_HEAT,
+    DWH_OFF_STATE: HVAC_MODE_OFF,
+}
+MAP_REVERSE_HVAC_MODES = dict(map(reversed, MAP_HVAC_MODES.items()))
+
+MAP_PRESET_MODES = {
+    PASS_APCDHW_MODE_ECO: PRESET_ECO,
+    PASS_APCDHW_MODE_COMFORT: PRESET_COMFORT,
+    PRESET_BOOST: PRESET_BOOST,
+    PASS_APCDWH_MODE_INTERNAL_SCHEDULING: CUSTOM_PRESET_PROG,
+    PASS_APCDHW_MODE_STOP: CUSTOM_PRESET_OFF,
+}
+MAP_REVERSE_PRESET_MODES = dict(map(reversed, MAP_PRESET_MODES.items()))
+
+
+class AtlanticPassAPCDHW(TahomaDevice, ClimateEntity):
+    """Representation of TaHoma IO Atlantic Electrical Heater."""
+
+    @property
+    def temperature_unit(self) -> str:
+        """Return the unit of measurement used by the platform."""
+        return TEMP_CELSIUS
+
+    @property
+    def supported_features(self) -> int:
+        """Return the list of supported features."""
+        return SUPPORT_PRESET_MODE | SUPPORT_TARGET_TEMPERATURE
+
+    @property
+    def min_temp(self) -> float:
+        """Return minimum temperature."""
+        return 30
+
+    @property
+    def max_temp(self) -> float:
+        """Return maximum temperature."""
+        return 65
+
+    @property
+    def preset_mode(self) -> Optional[str]:
+        """Return the current preset mode, e.g., home, away, temp."""
+        if self.select_state(IO_PASS_APCDWH_MODE_STATE) in [
+            PASS_APCDHW_MODE_ECO,
+            PASS_APCDWH_MODE_INTERNAL_SCHEDULING,
+            PASS_APCDHW_MODE_STOP,
+        ]:
+            return MAP_PRESET_MODES[self.select_state(IO_PASS_APCDWH_MODE_STATE)]
+
+        if self.select_state(CORE_BOOST_ON_OFF_STATE) == BOOST_ON_STATE:
+            return PRESET_BOOST
+
+        return PRESET_COMFORT
+
+    @property
+    def preset_modes(self) -> Optional[List[str]]:
+        """Return a list of available preset modes."""
+        return [*MAP_REVERSE_PRESET_MODES]
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set new preset mode."""
+        if preset_mode in [
+            PRESET_COMFORT,
+            PRESET_ECO,
+            CUSTOM_PRESET_OFF,
+            CUSTOM_PRESET_PROG,
+        ]:
+            presetModeToSet = MAP_REVERSE_PRESET_MODES[preset_mode]
```

Comment:
```suggestion
            preset_mode_to_set = MAP_REVERSE_PRESET_MODES[preset_mode]
```

Please use snake_case for your variables.